oauth2/src/main/java/com/okta/spring/boot/oauth/OktaOAuth2AutoConfig.java

```diff
@@ -65,4 +73,21 @@ OidcClientInitiatedLogoutSuccessHandler oidcLogoutSuccessHandler(OktaOAuth2Prope
     OAuth2UserService<OidcUserRequest, OidcUser> oidcUserService(Collection<AuthoritiesProvider> authoritiesProviders) {
         return new OktaOidcUserService(oAuth2UserService(authoritiesProviders), authoritiesProviders);
     }
+
+    @Bean
+    @ConditionalOnProperty(name="okta.oauth2.opaque", havingValue="true")
+    OpaqueTokenIntrospector oktaOpaqueTokenIntrospector(OktaOAuth2Properties oktaOAuth2Properties,
+                                                        OAuth2ResourceServerProperties oAuth2ResourceServerProperties) {
+        RestTemplate restTemplate = new RestTemplate(Arrays.asList(
+            new FormHttpMessageConverter(),
+            new StringHttpMessageConverter()));
+        restTemplate.setErrorHandler(new OAuth2ErrorResponseErrorHandler());
+        restTemplate.getInterceptors().add(new UserAgentRequestInterceptor());
```

Comment:
It's probably worth pulling his logic out into a common method as we have a few similar usages already.

Reply:
done